providers/coordinator.go

```diff
@@ -153,7 +153,7 @@ func (c *coordinator) unsafeRefreshRuntimes() {
 	var remaining []*Runtime
 	for i := range c.unprocessedRuntimes {
 		rt := c.unprocessedRuntimes[i]
-		if asset := rt.asset(); asset == nil || !c.unsafeSetAssetRuntime(asset, rt) {
+		if asset := rt.asset(); (asset == nil || !c.unsafeSetAssetRuntime(asset, rt)) && !rt.isClosed {
```

Comment:
nit:
can we check for `rtIsClosed` before even getting the asset? if closed we can just continue